### PR TITLE
First stab at retina-aware TUIImage.

### PR DESCRIPTION
**Pull request by lorenbrichter**
Make TUIImage/imageNamed: lazy, will reload different variants if needed when drawing based on the current scale factor of the destination context. Track the scale factor of the current destination context with a thread-specific global (so drawsInBackground works), set in TUIView/displayLayer.

### CODE REVIEW

File: lib/UIKit/TUIView.m [hunk 1]
```diff
@@ -286,6 +291,31 @@ - (CGContextRef)_CGContext
...
+	 Key is set up in +initialize
+	 Use TLS rather than a simple global so drawsInBackground should continue to work (views in the same process may be drawing destined for different windows on different screens with different scale factors).
+	 */
+	CGFloat *v = pthread_getspecific(TUICurrentContextScaleFactorTLSKey);
+	if(v)
+		return *v;
+	return 1.0;
+}
+
+static void BogusFunctionToSilenceStaticAnalyzer(void *x) { }
```
Comment (joshaber) [inline lib/UIKit/TUIView.m:306]: Hah, wat?

Comment (lorenbrichter) [inline lib/UIKit/TUIView.m:306]: BogusFunctionToSilenceStaticAnalyzer? The static analyzer complains that the malloc in SetCurrentContextScaleFactor isn't freed (it is, as part of the pthread_key_create destructor. Passing the pointer to a bogus function silences the warning.

Better way to do it? I looked for some kind of #pragma that would silence the analyzer, but couldn't find one.

Which reminds me to file a Radar on that...

Comment (lorenbrichter) [inline lib/UIKit/TUIView.m:306]: Scratch that, building clean without that workaround seems to have cleared up the issue. I'll submit a new pull request.

Comment (joshaber) [inline lib/UIKit/TUIView.m:306]: Ahhh...

So the really nasty but slightly less hacky way to do this:

``` objc
#ifndef __clang_analyzer__
// stuff that confuses the analyzer
#endif
```

Comment (joshaber) [inline lib/UIKit/TUIView.m:306]: FWIW in the future, you can still edit PR's after you've opened them.

Comment (lorenbrichter) [inline lib/UIKit/TUIView.m:306]: Oh sweet! Thanks for both tips.

**Pull request closed without merging**